src/librustc_borrowck/diagnostics.rs

```diff
@@ -286,6 +286,66 @@ You can read more about cell types in the API documentation:
 https://doc.rust-lang.org/std/cell/
 "##,
 
+E0389: r##"
+An attempt was made to mutate data using a non-mutable reference. This
+commonly occurs when attempting to assign to a reference of type
```

Comment:
Haha that sounds crazy to me-- that's why I just wrote the type out directly ("a non-mutable reference of a mutable reference" vs. `&(&mut T)`). But if you prefer, I'm happy to change it as I don't think it's going to greatly confuse anyone either way.